jwst/saturation/tests/test_saturation.py

```diff
@@ -0,0 +1,344 @@
+
+    # Create inputs, data, and saturation maps
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+    satvalue = 60000
+
+    data, satmap = setup_nrc_cube(ngroups, nrows, ncols)
+
+    # Add ramp values up to the saturation limit
+    data.data[0, 0, 500, 500] = 0
+    data.data[0, 1, 500, 500] = 20000
+    data.data[0, 2, 500, 500] = 40000
+    data.data[0, 3, 500, 500] = 60000
+    data.data[0, 4, 500, 500] = 62000   # Signal reaches saturation limit
+
+    # Set saturation value in the saturation model & DQ value for NO_SAT_CHECK
+    satmap.data[500, 500] = satvalue
+    satmap.dq[500, 500] = dqflags.pixel['NO_SAT_CHECK']
+
+    # Run the pipeline
+    output = do_correction(data, satmap)
+
+    # Make sure output GROUPDQ does not get flagged as saturated
+    # Make sure PIXELDQ is set to NO_SAT_CHECK
+    assert np.all(output.groupdq[0, :, 500, 500] != dqflags.group['SATURATED'])
+    assert output.pixeldq[500, 500] == dqflags.pixel['NO_SAT_CHECK']
+
+
+def test_nans_in_mask(setup_nrc_cube):
+    '''Check that pixels in the reference files that have value NaN are not
+       flagged as saturated in the data and that in the PIXELDQ array the
+       pixel is set to NO_SAT_CHECK.'''
+
+    # Create inputs, data, and saturation maps
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+    huge_num = 100000.
+
+    data, satmap = setup_nrc_cube(ngroups, nrows, ncols)
+
+    # Add ramp values up to the saturation limit
+    data.data[0, 0, 500, 500] = 0
+    data.data[0, 1, 500, 500] = 20000
+    data.data[0, 2, 500, 500] = 40000
+    data.data[0, 3, 500, 500] = 60000
+    data.data[0, 4, 500, 500] = 62000
+
+    # Set saturation value for pixel to NaN
+    satmap.data[500, 500] = np.nan
+
+    # Run the pipeline
+    correct_for_NaN(satmap.data, satmap.dq)
+    output = do_correction(data, satmap)
+
+    # Check that NaN reference value gets reset to HUGE_NUM
+    # Check that reference DQ is set to NO_SAT_CHECK
+    # Check that output GROUPDQ is not flagged as saturated
+    # Check that output PIXELDQ is set to NO_SAT_CHECK
+    assert satmap.data[500, 500] == huge_num
+    assert satmap.dq[500, 500] == dqflags.pixel['NO_SAT_CHECK']
+    assert np.all(output.groupdq[0, :, 500, 500] != dqflags.group['SATURATED'])
+    assert output.pixeldq[500, 500] == dqflags.pixel['NO_SAT_CHECK']
+
+
+def test_full_step(setup_nrc_cube):
+    '''Test full run of the SaturationStep.'''
+
+    # Create inputs, data, and saturation maps
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+
+    data, satmap = setup_nrc_cube(ngroups, nrows, ncols)
+
+    # Add ramp values up to the saturation limit
+    data.data[0, 0, 500, 500] = 0
+    data.data[0, 1, 500, 500] = 20000
+    data.data[0, 2, 500, 500] = 40000
+    data.data[0, 3, 500, 500] = 60000   # Signal reaches saturation limit
+    data.data[0, 4, 500, 500] = 62000
+
+    # Get reference file
+    reffile = "/grp/crds/cache/references/jwst/jwst_nircam_saturation_0064.fits"
```

Comment:
This line is causing this test to fail.  None of our CI testing infrastructure has access to central store, so instead of setting `override_saturation`, perhaps one could just specify the required header keywords in `data` so that the 0064 reffile gets selected.

The current rmap for NIRCam saturation reffile

https://jwst-crds.stsci.edu

Indicates that the following keywords are required:
```
data.meta.instrument.detector = 'NRCA1'
data.meta.observation.date = '2017-10-01'
data.meta.observation.time = '00:00:00'
```
and it will get selected.  I don't know if this is too fragile, but it will work.  One may want to first assert that the reffile is the correct one you're expecting too.
```
assert output.meta.ref_file.saturation.name == 'crds://jwst_nircam_saturation_0064.fits'
```